cpp/src/io/shp/polygon_shapefile_reader.cpp

```diff
@@ -53,15 +53,16 @@ cudf::size_type read_ring(OGRLinearRing const& ring,
 cudf::size_type read_polygon(OGRPolygon const& polygon,
                              std::vector<int>& ring_lengths,
                              std::vector<double>& xs,
-                             std::vector<double>& ys)
+                             std::vector<double>& ys,
+                             const cuspatial::winding_order ring_order)
 {
-  auto num_vertices = read_ring(*(polygon.getExteriorRing()), xs, ys);
+  auto num_vertices = read_ring(*(polygon.getExteriorRing()), xs, ys, ring_order);
   ring_lengths.push_back(num_vertices);
 
   cudf::size_type num_interior_rings = polygon.getNumInteriorRings();
 
   for (cudf::size_type i = 0; i < num_interior_rings; i++) {
```

Comment:
Where is the order flipping happening? At each nested level the order should be reversed.

Reply:
Order flipping occurs in the interior rings as well in the next couple of [lines](https://github.com/rapidsai/cuspatial/pull/609/files/a971db87aa08b253764f795844c11a3d305e1f74..dae0fe361808676cb8456994c90121c51468a640#diff-ff99c11a9d678705f3f645969a9e6d8d684e5c8e0f3ec589967bcc29fca58a48R65)

Reply:
Seems to be it's just passed down through the rings without alternating orders.

Reply:
The interior rings are assumed to be opposite winding of the outer ring. The argument only specifies the winding for the outer ring, so if we wind the outer ring reverse, all the children rings get reversed as well. Otherwise neither are changed.